riptable/rt_str.py

```diff
@@ -915,5 +917,81 @@ def char(self, position: Union[int, List[int], np.ndarray]):
     nb_char_par = _njit_par(_nb_char)
 
 
+MIN_INT = np.iinfo(np.int32).min
+
+
+def _populate_wrappers(cls):
+    """
+    Decorator for the CatString class which populates the string methods and sets the defaults for
+    filtered value fills.
+    """
+    for method, filter_value_fill in [
+        ('upper', None),
+        ('lower', None),
+        ('reverse', None),
+        ('removetrailing', None),
+        ('strlen', MIN_INT),
+        ('index_any_of', MIN_INT),
+        ('index', MIN_INT),
+        ('contains', False),
+        ('startswith', False),
+        ('endswith', False),
+        ('regex_match', False),
+        ('substr', None),
+        ('char', None)
+    ]:
+        setattr(cls, method, cls._build_method(method, filter_value_fill))
+    return cls
+
+
+@_populate_wrappers
+class CatString:
+    """
+    Provides access to FAString methods for Categoricals.
+    All string methods are wrappers of the FAString equivalent with
+    categorical re-expansion and option for how to fill filtered elements.
+    """
+    def __init__(self, cat):
+        from .rt_categorical import CategoryMode
+        self.cat = cat
+        if cat.isenum:
+            raise ValueError(f"Could not use str in enum mode.  Email for help")
+        elif cat.issinglekey:
+            string_list = cat.category_array
+            self.fastring = FAString(string_list)
+
+        elif cat.ismultikey:
+            raise ValueError(f"Could not use .str in multikey mode.")
+        else:
+            raise ValueError(f"Could not use .str in {CategoryMode(cat.category_mode).name}.")
+
+    @classmethod
+    def _build_method(cls, method, filter_value_fill):
+        """
+        General purpose factory for FAString wrappers.
+        """
+        func = getattr(FAString, method)

```

Comment:
getattr is intended to be used for attributes (although it can work here for most cases). To ensure you are getting a defined method you can use the inspect module to inspect.getmembers of the class and filter down the methods (using inspect.ismethod) and checking if it is a property as you have with the isinstance check.
https://docs.python.org/3/library/inspect.html